Kudu.Core/SiteExtensions/SiteExtensionStatus.cs

```diff
@@ -117,6 +119,11 @@ public bool IsTerminalStatus()
                 || string.Equals(Constants.SiteExtensionProvisioningStateCanceled, ProvisioningState, StringComparison.OrdinalIgnoreCase);
         }
 
+        public bool IsRequireRestart()
+        {
+            return IsSiteExtensionRequireRestart(this, _rootPath);
```

Comment:
Why is `IsSiteExtensionRequireRestart` static? If it takes the instance in, why not make it instance?